bigchaindb/backend/mongodb/query.py

```diff
@@ -153,13 +153,17 @@ def get_spent(conn, transaction_id, output):
     cursor = conn.run(
         conn.collection('bigchain').aggregate([
             {'$match': {
-                'block.transactions.inputs.fulfills.txid': transaction_id,
-                'block.transactions.inputs.fulfills.output': output
+                'block.transactions.inputs.fulfills': {
```

Comment:
I don't think there is necessarily a problem. We just need to use a better query